### DISCUSSION TITLE
How can I build a query that sorts a collection by a composite index but in descending order

### DISCUSSION THREAD

**Discussion by Sockitos** [isar/isar#625] [Help] [answered]
Hi,

I'm using a collection to store episodes that have a composite index with **showId**, **season** and **episode**.
By default, the collection is in ascending order according to that index but I need to make a query that returns the last episode for all **showIds** in the collection.
For that I would need a query like this:
`final query = collection.where().sortBySeasonDesc().sortByEpisodeDesc().distinctByShowId().build();`
Unfortunately, the generated methods don't allow to have a sortBy after another sortBy.

Is there currently a way to do this query with the generated methods, and if not how can I do it with a dynamic query?

Thanks!

**Comment by Sockitos** [comment, marked as answer]
Already found out the correct way to do it. Didn't know there were **thenBy** methods.
The correct query would be:
`final query = collection.where().sortBySeasonDesc().thenByEpisodeDesc().distinctByShowId().build();`

**Reply by simc** [reply]
That works but it doesn't use the index. Use the following instead:
`collection.where().anyShowIdSeasonEpisode().distinctByShowId().build()`

**Reply by Sockitos** [reply]
I'm a bit confused.
According to the docs, .where() queries use indexes. Why is this not the case in my query?
> Sorting a lot of results can be an expensive operation. Luckily, we can again use indexes an make our query lighning fast even if we need to sort a million objects.

Thank you for your suggestion, it also works but you forgot to add `sort: Sort.desc`.

**Reply by simc** [reply]
I agree the query builder and docs are a bit misleading there I'll try to fix it. All the sorting methods do not use indexes.
